wrappers/golang/README.md

```diff
@@ -0,0 +1,107 @@
+# Golang Bindings
+
+In order to build the underlying ICICLE libraries you should run the build script found [here](./build.sh).
+
+Build script USAGE
+
+```sh
+./build.sh [-curve=<curve>] [-field=<field>] [-hash=<hash>] [-cuda_version=<version>] [-skip_msm] [-skip_ntt] [-skip_g2] [-skip_ecntt] [-skip_fieldext]
+
+curve - The name of the curve to build or "all" to build all supported curves
+field - The name of the field to build or "all" to build all supported fields
+-skip_msm - Optional - build with MSM disabled
+-skip_ntt - Optional - build with NTT disabled
+-skip_g2 - Optional - build with G2 disabled 
+-skip_ecntt - Optional - build with ECNTT disabled
+-skip_fieldext - Optional - build without field extension
+-help - Optional - Displays usage information
+```
+
+To build ICICLE libraries for all supported curves with G2 and ECNTT enabled.
+
+```sh
+./build.sh -curve=all
+```
+
+If you wish to build for a specific curve, for example bn254, without G2 or ECNTT enabled.
+
+```sh
+./build.sh -curve=bn254 -skip_g2 -skip_ecntt
+```
+
+## Supported curves, fields and operations
+
+### Supported curves and operations
+
+| Operation\Curve | bn254 | bls12_377 | bls12_381 | bw6-761 | grumpkin |
+| --- | :---: | :---: | :---: | :---: | :---: |
+| MSM | ✅ | ✅ | ✅ | ✅ | ✅ |
+| G2  | ✅ | ✅ | ✅ | ✅ | ❌ |
+| NTT | ✅ | ✅ | ✅ | ✅ | ❌ |
+| ECNTT | ✅ | ✅ | ✅ | ✅ | ❌ |
+| VecOps | ✅ | ✅ | ✅ | ✅ | ✅ |
+| Polynomials | ✅ | ✅ | ✅ | ✅ | ❌ |
+
+### Supported fields and operations
+
+| Operation\Field | babybear |
+| --- | :---: |
+| VecOps | ✅ |
+| Polynomials | ✅ |
+| NTT | ✅ |
+| Extension Field | ✅ |
+
+## Running golang tests
+
+To run the tests for curve bn254.
+
+```sh
+go test ./wrappers/golang_v3/curves/bn254/tests -count=1 -v
+```
+
+To run all the tests in the golang bindings
+
+```sh
+go test ./... -count=1 -v
+```
+
+## How do Golang bindings work?
+
+The libraries produced from the CUDA code compilation are used to bind Golang to ICICLE's CUDA code.
```

Comment:
We shouldn't specify CUDA here as the bindings work for the CPU backend as well

Reply:
@jeremyfelder can fix?